packages/commonwealth/client/scripts/controllers/server/rules.ts

```diff
@@ -0,0 +1,154 @@
+import RuleStore from 'stores/RulesStore';
+import app from 'state';
+import $ from 'jquery';
+import { Rule } from 'models';
+
+export const ruleModelFromServer = (ruleFromDb) => {
+  const { id, chain_id, rule, created_at, updated_at } = ruleFromDb;
+
+  return new Rule({
+    id,
+    chainId: chain_id,
+    rule,
+    createdAt: created_at,
+    updatedAt: updated_at,
+  });
+};
+
+class RulesController {
+  private _rulesStore = new RuleStore();
+
+  // Checks if a provided address passes a given rule.
+  // Agnostic to chain; this will work even if the rule is not in the store
+  public async addressPassesRule({
+    rule_id,
+    address,
+  }: {
+    rule_id: number;
+    address: string;
+  }): Promise<boolean> {
+    try {
+      const res = await $.post(`${app.serverUrl()}/checkAddressAgainstRule`, {
+        rule_id,
+        address,
+        jwt: app.user.jwt,
+      });
+      return res.result.addressIsValidUnderRule;
+    } catch (e) {
+      console.log(e);
+      throw new Error('Unable to check address against provided rule');
+    }
+  }
+
+  public async refresh() {
+    // First clear the store
+    this._rulesStore.clear();
+
+    // Populate store with rules corresponding to current community
+    try {
+      const res = await $.post(`${app.serverUrl()}/getRules`, {
+        chain_id: app.activeChainId() ?? null,
+        jwt: app.user.jwt,
+      });
+
+      if (res.result.rules.length > 0) {
+        await res.result.rules.forEach((rule) => {
+          this._rulesStore.add(ruleModelFromServer(rule));
+        });
+      }
+    } catch (e) {
+      console.log(e);
+      throw new Error('Failed to refresh rules');
+    }
+  }
+
+  public async createRule({
+    chain_id,
+    ruleSchema,
+  }: {
+    chain_id: string;
+    ruleSchema: any;
```

Comment:
We may want to move some of the internal rules function to `shared` or `common-common` so we can access proper types / checking functions on the client side.

Reply:
@jnaviask @alexyoung23j Should this happen in this PR so we can merge this?